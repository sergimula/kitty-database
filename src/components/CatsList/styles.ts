import { Theme } from '@material-ui/core/styles'

const styles = (theme: Theme) => ({
  root: {
    flexGrow: 1,
    maxWidth: 752,
  },
  demo: {
    backgroundColor: theme.palette.background.paper,
    width: '100%',
  },
  title: {
    margin: theme.spacing(4, 0, 2),
  },
})

export default styles
