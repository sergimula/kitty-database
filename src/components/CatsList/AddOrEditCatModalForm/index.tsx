import React, { useEffect, useReducer } from 'react'
import { get, head, toInteger } from 'lodash'
import {
  Button,
  Dialog,
  DialogTitle,
  FormControl,
  Grid,
  InputLabel,
  makeStyles,
  Select,
  TextField,
} from '@material-ui/core'

import { Cat, CatRace, PLACE_KITTEN_BASE_URL } from '../../../constants'
import * as ducks from './ducks'
import styles from './styles'

const useStyles = makeStyles(styles)

export interface Props {
  addCatToDB?: (cat: Cat) => void
  catsRaces: Array<CatRace>
  catToEdit?: Cat
  editCat?: (cat: Cat) => void
  isOpen: boolean
  setIsModalFormOpen: (value: boolean) => void
}

const AddCatModalForm = (props: Props) => {
  const classes = useStyles()
  const { catsRaces, isOpen, setIsModalFormOpen } = props
  const [state, dispatch] = useReducer(ducks.reducer, ducks.getInitialState())

  useEffect(() => {
    if (isOpen && props.catToEdit) {
      dispatch({ type: ducks.EDIT_CAT, value: props.catToEdit })
    }
  }, [isOpen, props.catToEdit])

  return (
    <Dialog
      onClose={() => {
        setIsModalFormOpen(false)
        dispatch({ type: ducks.RESET_FORM })
      }}
      open={isOpen}
    >
      <DialogTitle>
        {props.catToEdit
          ? `Editing ${props.catToEdit.name}`
          : 'Add cat to database'}
      </DialogTitle>
      <form
        className={classes.addCatForm}
        name='add-cat-form'
        onSubmit={(event) => {
          event.preventDefault()

          if (props.addCatToDB) {
            props.addCatToDB({
              name: state.name,
              description: state.description,
              race: get(
                head(catsRaces.filter((race) => race.id === state.race_id)),
                'name',
                state.race
              ),
              race_id: state.race_id,
              image: `${state.image}/${state.imageWidth}/${state.imageHeight}`,
            })
          }

          if (props.editCat) {
            props.editCat({
              id: state.id,
              name: state.name,
              description: state.description,
              race: get(
                head(catsRaces.filter((race) => race.id === state.race_id)),
                'name',
                state.race
              ),
              race_id: state.race_id,
              image: `${PLACE_KITTEN_BASE_URL}/${state.imageWidth}/${state.imageHeight}`,
            })
          }

          setIsModalFormOpen(false)
        }}
      >
        <React.Fragment>
          <Grid container justify='center' spacing={2}>
            <Grid item xs={12}>
              <TextField
                fullWidth
                label='Cat name'
                name='name'
                onChange={(event) =>
                  dispatch({
                    type: ducks.SET_FIELD_VALUE,
                    field: event.target.name,
                    value: event.target.value,
                  })
                }
                required
                value={state.name}
                variant='outlined'
              />
            </Grid>
            <Grid item xs={12}>
              <TextField
                fullWidth
                label='Cat description'
                name='description'
                onChange={(event) =>
                  dispatch({
                    type: ducks.SET_FIELD_VALUE,
                    field: event.target.name,
                    value: event.target.value,
                  })
                }
                value={state.description}
                variant='outlined'
              />
            </Grid>
            <Grid item xs={6}>
              <TextField
                fullWidth
                label='Cat image width'
                name='imageWidth'
                onChange={(event) =>
                  dispatch({
                    type: ducks.SET_FIELD_VALUE,
                    field: event.target.name,
                    value: event.target.value,
                  })
                }
                required
                type='number'
                value={state.imageWidth}
                variant='outlined'
              />
            </Grid>
            <Grid item xs={6}>
              <TextField
                fullWidth
                label='Cat image height'
                name='imageHeight'
                onChange={(event) =>
                  dispatch({
                    type: ducks.SET_FIELD_VALUE,
                    field: event.target.name,
                    value: event.target.value,
                  })
                }
                required
                type='number'
                value={state.imageHeight}
                variant='outlined'
              />
            </Grid>
            <Grid item xs={12}>
              <FormControl className={classes.select} variant='outlined'>
                <InputLabel htmlFor='add-cat-race-selector'>Race</InputLabel>
                <Select
                  inputProps={{
                    name: 'race',
                    id: 'add-cat-race-selector',
                  }}
                  label='Race'
                  native
                  onChange={(event) =>
                    dispatch({
                      type: ducks.SET_FIELD_VALUE,
                      field: 'race_id',
                      value: toInteger(event.target.value),
                    })
                  }
                  value={state.race_id}
                >
                  {catsRaces.map((race: CatRace) => (
                    <option key={race.id} value={race.id}>
                      {race.name}
                    </option>
                  ))}
                </Select>
              </FormControl>
            </Grid>
          </Grid>
          <Grid container justify='flex-end' spacing={2}>
            <Grid item xs={3}>
              <Button
                fullWidth
                onClick={() => {
                  setIsModalFormOpen(false)
                  dispatch({ type: ducks.RESET_FORM })
                }}
                size='large'
              >
                Cancel
              </Button>
            </Grid>
            <Grid item xs={3}>
              <Button
                color='primary'
                fullWidth
                size='large'
                type='submit'
                variant='contained'
              >
                {props.catToEdit ? `Edit cat` : 'Add cat'}
              </Button>
            </Grid>
          </Grid>
        </React.Fragment>
      </form>
    </Dialog>
  )
}

export default AddCatModalForm
