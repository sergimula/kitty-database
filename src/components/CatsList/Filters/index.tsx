import React, { useState, useEffect } from 'react'
import { isEmpty, toLower, toInteger } from 'lodash'
import {
  FormControl,
  Grid,
  InputLabel,
  makeStyles,
  Select,
  TextField,
} from '@material-ui/core'

import { Cat, CatRace } from 'src/constants'
import styles from './styles'

const useStyles = makeStyles(styles)

interface Props {
  catsList: Array<Cat>
  catsRaces: Array<CatRace>
  setCatsList: Function
}

const Filters = (props: Props) => {
  const [searchValue, setSearchValue] = useState<string>('')
  const [catRaceId, setCatRaceId] = useState<number>(0)
  const classes = useStyles()

  useEffect(() => {
    //Filter by name & race
    if (!isEmpty(searchValue) && catRaceId !== 0) {
      const filteredCatsListByNameAndRace = props.catsList.filter(
        (cat: Cat) => {
          return (
            toLower(cat.name).includes(searchValue.toLowerCase()) &&
            cat.race_id === catRaceId
          )
        }
      )
      return props.setCatsList(filteredCatsListByNameAndRace)
    }
    //Filter by name
    if (!isEmpty(searchValue)) {
      const filteredCatsListByName = props.catsList.filter((cat: Cat) => {
        return toLower(cat.name).includes(searchValue.toLowerCase())
      })
      return props.setCatsList(filteredCatsListByName)
    }
    //Filter by race
    if (catRaceId !== 0) {
      const filteredCatsListByRace = props.catsList.filter((cat: Cat) => {
        return cat.race_id === catRaceId
      })
      return props.setCatsList(filteredCatsListByRace)
    }
    //Set default catsList value
    if (isEmpty(searchValue) && catRaceId === 0) {
      props.setCatsList(props.catsList)
    }
  }, [catRaceId, props.catsList, props.setCatsList, searchValue])

  return (
    <Grid
      container
      className={classes.filtersContainer}
      justify='center'
      spacing={2}
    >
      <Grid item xs={9}>
        <TextField
          fullWidth
          label='Search Cat by name'
          onChange={(event) => {
            setSearchValue(event.target.value)
          }}
          value={searchValue}
          variant='outlined'
        />
      </Grid>
      <Grid item xs={3}>
        <FormControl className={classes.select} variant='outlined'>
          <InputLabel htmlFor='race-selector'>Race</InputLabel>
          <Select
            inputProps={{
              name: 'race',
              id: 'race-selector',
            }}
            label='Race'
            native
            onChange={(event) => setCatRaceId(toInteger(event.target.value))}
            value={catRaceId}
          >
            <option value={0}>All</option>
            {props.catsRaces.map((race: CatRace) => (
              <option key={race.id} value={race.id}>
                {race.name}
              </option>
            ))}
          </Select>
        </FormControl>
      </Grid>
    </Grid>
  )
}

export default Filters
