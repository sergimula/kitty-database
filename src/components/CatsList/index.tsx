import React, { useState, useEffect } from 'react'
import {
  Avatar,
  Button,
  Card,
  CardActions,
  Grid,
  IconButton,
  List,
  ListItem,
  ListItemAvatar,
  ListItemSecondaryAction,
  ListItemText,
  makeStyles,
  Typography,
} from '@material-ui/core'
import DeleteIcon from '@material-ui/icons/Delete'
import EditIcon from '@material-ui/icons/Edit'
import InfoIcon from '@material-ui/icons/Info'
import PetsIcon from '@material-ui/icons/Pets'

import { Cat, CatRace, PLACE_KITTEN_BASE_URL } from '../../constants'
import Filters from './Filters'
import AddOrEditCatModalForm from './AddOrEditCatModalForm'
import styles from './styles'

const useStyles = makeStyles(styles)

interface Props {
  addCat: (cat: Cat) => void
  cats: Array<Cat>
  catsRaces: Array<CatRace>
  editCat: (cat: Cat) => void
  removeCat: (cat: Cat) => void
}

const CatsList = (props: Props) => {
  const [catsList, setCatsList] = useState<Array<Cat>>([])
  const [isAddCatModalFormOpen, setIsAddCatModalFormOpen] = useState<boolean>(
    false
  )
  const [isEditCatModalFormOpen, setIsEditCatModalFormOpen] = useState<boolean>(
    false
  )
  const [showCat, setShowCat] = useState<Cat | undefined>(undefined)
  const [catToEdit, setCatToEdit] = useState<Cat>({
    name: '',
    description: '',
    image: `${PLACE_KITTEN_BASE_URL}/200/200`,
    race: 'European',
    race_id: 1,
  })
  const classes = useStyles()

  useEffect(() => {
    setCatsList(props.cats)
  }, [props.cats])

  return (
    <Card className={classes.root}>
      <Grid
        alignItems='center'
        container
        direction='column'
        justify='center'
        spacing={2}
      >
        <Typography
          align='center'
          gutterBottom
          variant='h5'
          component='h2'
          className={classes.title}
        >
          Cats Database
        </Typography>
        <Grid item xs={12} className={classes.demo}>
          {catsList.length === 0 ? (
            <Typography align='center' variant='h6' className={classes.title}>
              No cats matching
            </Typography>
          ) : (
            <List dense={false}>
              {catsList.map((cat: Cat) => (
                <React.Fragment key={cat.id}>
                  <ListItem>
                    <ListItemAvatar>
                      <Avatar>
                        <PetsIcon />
                      </Avatar>
                    </ListItemAvatar>
                    <ListItemText primary={cat.name} />
                    <ListItemSecondaryAction>
                      <IconButton
                        edge='end'
                        aria-label='delete'
                        onClick={() => {
                          setShowCat(cat)
                        }}
                      >
                        <InfoIcon />
                      </IconButton>
                      <IconButton
                        edge='end'
                        aria-label='delete'
                        onClick={() => {
                          setCatToEdit(cat)
                          setIsEditCatModalFormOpen(true)
                        }}
                      >
                        <EditIcon />
                      </IconButton>
                      <IconButton
                        edge='end'
                        aria-label='delete'
                        onClick={() => props.removeCat(cat)}
                      >
                        <DeleteIcon />
                      </IconButton>
                    </ListItemSecondaryAction>
                  </ListItem>
                  {showCat && cat.id === showCat.id && (
                    <Grid container justify='center' spacing={1}>
                      <Grid item>
                        <img alt={showCat.name} src={showCat.image} />
                      </Grid>
                      <Grid item>
                        <Typography>{showCat.description}</Typography>
                      </Grid>
                    </Grid>
                  )}
                </React.Fragment>
              ))}
            </List>
          )}
        </Grid>
      </Grid>
      <CardActions>
        <Grid container direction='column'>
          <Grid container justify='flex-end'>
            <Button
              size='large'
              color='secondary'
              variant='contained'
              endIcon={<PetsIcon />}
              onClick={() => setIsAddCatModalFormOpen(true)}
            >
              Add new cat
            </Button>
            <AddOrEditCatModalForm
              addCatToDB={props.addCat}
              catsRaces={props.catsRaces}
              isOpen={isAddCatModalFormOpen}
              setIsModalFormOpen={setIsAddCatModalFormOpen}
            />
          </Grid>
          <Filters
            catsList={props.cats}
            catsRaces={props.catsRaces}
            setCatsList={setCatsList}
          />
        </Grid>
      </CardActions>
      <AddOrEditCatModalForm
        editCat={props.editCat}
        catsRaces={props.catsRaces}
        catToEdit={catToEdit}
        isOpen={isEditCatModalFormOpen}
        setIsModalFormOpen={setIsEditCatModalFormOpen}
      />
    </Card>
  )
}

export default CatsList
