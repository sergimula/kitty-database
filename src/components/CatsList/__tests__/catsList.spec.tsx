import React from 'react'
import { head } from 'lodash'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import {
  Button,
  Dialog,
  IconButton,
  Typography
} from '@material-ui/core'

import CatsList from '../index'
import * as stubs from './stubs'

describe('CatsList', () => {
  it('should match CatsList', () => {
    const props = stubs.catsListProps()
    const wrapper = renderer.create(<CatsList {...props} />)

    expect(wrapper).toMatchSnapshot()
  })

  describe('when categories are empty', () => {
    it('should render no cats matching text', () => {
      const wrapper = mount(
        <CatsList {...stubs.catsListPropsEmptyList} />
      )

      expect(wrapper.find(Typography).at(1).text()).toBe('No cats matching')
    })
  })

  describe('when user hits delete button', () => {
    const props = stubs.catsListProps()
    const wrapper = mount(<CatsList {...props} />)
    const cat = head(stubs.cats)

    wrapper.find(IconButton).at(2).simulate('click')

    expect(props.removeCat).toHaveBeenCalledWith(cat)
  })

  describe('AddOrEditCatModalForm', () => {
    describe('when user hits Add Cat button', () => {
      it('should open Add Cat Modal form', () => {
        const props = stubs.catsListProps()
        const wrapper = mount(<CatsList {...props} />)

        wrapper.find(Button).at(0).simulate('click')
        wrapper.update()
        expect(wrapper.find(Dialog).at(0).props().open).toBe(true)
      })
    })

    describe('when user adds a new cat', () => {
      it('should call addCat function', () => {
        const props = stubs.catsListProps()
        const wrapper = mount(<CatsList {...props} />)

        wrapper.find(Button).at(0).simulate('click')
        wrapper.update()
        wrapper.find(Dialog).at(0).find('input[name="name"]').simulate('change', {
          target: { value: stubs.testCastToAdd.name, name: 'name' }
        })
        wrapper.find(Dialog).at(0).find('input[name="description"]').simulate('change', {
          target: { value: stubs.testCastToAdd.description, name: 'description' }
        })
        wrapper.find(Dialog).at(0).find('input[name="imageHeight"]').simulate('change', {
          target: { value: stubs.testCastToAdd.imageHeight, name: 'imageHeight' }
        })
        wrapper.find(Dialog).at(0).find('input[name="imageWidth"]').simulate('change', {
          target: { value: stubs.testCastToAdd.imageWidth, name: 'imageWidth' }
        })

        wrapper.find(Dialog).at(0).find(Button).at(1).simulate('submit')

        expect(props.addCat).toHaveBeenCalledWith(stubs.testCatAdded)
      })
    })

    describe('when user cancels the action', () => {
      it('should close Modal form', () => {
        const props = stubs.catsListProps()
        const wrapper = mount(<CatsList {...props} />)

        wrapper.find(Button).at(0).simulate('click')
        wrapper.update()
        wrapper.find(Dialog).at(0).find(Button).at(0).simulate('click')
        wrapper.update()
        expect(wrapper.find(Dialog).at(0).props().open).toBe(false)
      })
    })

    describe('when user hits edit cat button', () => {
      it('should open Edit Cat Modal Form with cat data', () => {
        const props = stubs.catsListProps()
        const wrapper = mount(<CatsList {...props} />)
        const cat = head(stubs.cats)

        wrapper.find(IconButton).at(1).simulate('click')
        wrapper.update()
        expect(wrapper.find(Dialog).at(1).props().open).toBe(true)
        expect(wrapper.find(Dialog).at(1).find('input[name="name"]').props().value).toBe(cat.name)
        expect(wrapper.find(Dialog).at(1).find('input[name="description"]').props().value).toBe(cat.description)
        expect(wrapper.find(Dialog).at(1).find('[name="race_id"]').at(0).props().value).toBe(cat.race_id)
      })
    })
  })
})
