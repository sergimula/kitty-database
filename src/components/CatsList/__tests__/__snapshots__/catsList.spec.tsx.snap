// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatsList should match CatsList 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <h2
      className="MuiTypography-root makeStyles-title-3 MuiTypography-h5 MuiTypography-gutterBottom MuiTypography-alignCenter"
    >
      Cats Database
    </h2>
    <div
      className="MuiGrid-root makeStyles-demo-2 MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h6
        className="MuiTypography-root makeStyles-title-3 MuiTypography-h6 MuiTypography-alignCenter"
      >
        No cats matching
      </h6>
    </div>
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-flex-end"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-containedSizeLarge MuiButton-sizeLarge"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Add new cat
            <span
              className="MuiButton-endIcon MuiButton-iconSizeLarge"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="4.5"
                  cy="9.5"
                  r="2.5"
                />
                <circle
                  cx="9"
                  cy="5.5"
                  r="2.5"
                />
                <circle
                  cx="15"
                  cy="5.5"
                  r="2.5"
                />
                <circle
                  cx="19.5"
                  cy="9.5"
                  r="2.5"
                />
                <path
                  d="M17.34 14.86c-.87-1.02-1.6-1.89-2.48-2.91-.46-.54-1.05-1.08-1.75-1.32-.11-.04-.22-.07-.33-.09-.25-.04-.52-.04-.78-.04s-.53 0-.79.05c-.11.02-.22.05-.33.09-.7.24-1.28.78-1.75 1.32-.87 1.02-1.6 1.89-2.48 2.91-1.31 1.31-2.92 2.76-2.62 4.79.29 1.02 1.02 2.03 2.33 2.32.73.15 3.06-.44 5.54-.44h.18c2.48 0 4.81.58 5.54.44 1.31-.29 2.04-1.31 2.33-2.32.31-2.04-1.3-3.49-2.61-4.8z"
                />
              </svg>
            </span>
          </span>
        </button>
      </div>
      <div
        className="MuiGrid-root makeStyles-filtersContainer-6 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-center"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink={false}
            >
              Search Cat by name
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-10"
                >
                  <span>
                    Search Cat by name
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
        >
          <div
            className="MuiFormControl-root makeStyles-select-7"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
              data-shrink={true}
              htmlFor="race-selector"
            >
              Race
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              onClick={[Function]}
            >
              <select
                aria-invalid={false}
                className="MuiSelect-root MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                id="race-selector"
                multiple={false}
                name="race"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                value={0}
              >
                <option
                  value={0}
                >
                  All
                </option>
                <option
                  value={1}
                >
                  European
                </option>
                <option
                  value={2}
                >
                  Siamese
                </option>
                <option
                  value={3}
                >
                  Siberian
                </option>
                <option
                  value={4}
                >
                  Persian
                </option>
              </select>
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-10 PrivateNotchedOutline-legendNotched-11"
                >
                  <span>
                    Race
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
